lib/api-v1.js

```diff
@@ -566,30 +567,31 @@ api.declare({
 }, async function(req, res) {
   let token = req.params.token;
 
-  try {
-    let secret = await this.Secret.load({
-      token: token,
-    });
+  let secret = await this.Secret.load({
+    token: token,
+  }, true);
 
-    return res.reply({
-      data: secret.secrets,
-      scopes: secret.scopes,
-      credentials: taskcluster.createTemporaryCredentials({
-        scopes: [
-          `assume:worker-type:${this.provisionerId}/${secret.workerType}`,
-          'assume:worker-id:*',
-        ],
-        expiry: taskcluster.fromNow('96 hours'),
-        credentials: this.credentials,
-      }),
-    });
-  } catch (err) {
-    if (err.code === 'ResourceNotFound') {
-      res.status(404).end();
-    } else {
-      throw err;
-    }
+  if (!secret || secret.expiration < new Date()) {
+    return res.reportError('ResourceNotFound', 'No such secret', {});
   }
+
+  // remove the token so that it can't be used repeatedly
+  await this.Secret.remove({
```

Comment:
Should we also be deleting those that are expired? Not a big deal either way.

Reply:
They won't work with this change, so I'm not going to mess with it more than necessary.